src/main/scala/karazin/scala/users/group/week1/homework/Homework.scala

```diff
@@ -40,26 +40,135 @@ object Homework :
 
   object `Boolean Operators` :
 
-    def not(b: Boolean): Boolean = ???
+    def not(b: Boolean): Boolean = if b then false else true
 
-    def and(left: Boolean, right: Boolean): Boolean = ???
+    def and(left: Boolean, right: Boolean): Boolean =
+                            // it's another variant to do this,
+                            // but I prefer more explicitness and readability
 
-    def or(left: Boolean, right: Boolean): Boolean = ???
+      if left then          // if left then
+        if right then true  //    if right then right
+        else false          //    else right
+      else false            // else left
+
+    def or(left: Boolean, right: Boolean): Boolean = if left then true else right

```

Comment:
Nice! Simplify `and` like you did it here.

This is quite good solution but in the requirements it's said that and should works like ||. In Your case `right` always is evaluated. I'd like you to add an additional test for
or(and, throw new Exception()) -- if `or` works as `||` then `Exception` won't be thrown. Hint: remind about "call by value strategy".

Reply:
Why `right` always is evaluated? I see that right is evaluated only when `left` is false and we need to evaluate `right`. Am I wrong?

Reply:
So I understand my mistake.